source/fundamentals/connection/tuning.txt

```diff
@@ -0,0 +1,166 @@
+=================
+Connection Storms
+=================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. _connection-tuning-java:
+
+Overview
+--------
+
+In this guide, you can learn about **connection storms** and settings that
+can help avoid them.
+
+Connection Storms
+-----------------
+
+A **connection storm** is an event in which application servers
+start an increasing number of connection requests, but the host server
+CPU is unable to schedule them.
+
+.. example::
+
+   Consider an application that uses MongoDB, and is served from an
+   auto scaling service. A sudden spike in traffic could cause the scaling
+   service to stand up many new application instances. This sudden burst
+   in application instances attempting to establish connections to MongoDB
+   is one example of a connection storm.
+
+Common Causes and Effects
++++++++++++++++++++++++++
+
+Common causes for connection storms in MongoDB include:
+
+- Incorrect application behavior
+- Network latency
+- Insufficient hardware for the workload
+- Failover events
+
+In MongoDB, connection storms can have multiple effects. For example:
+
+- The high volume of connection requests leads to latency in processing.
+- Replication connections between nodes may be blocked or slowed, so replication
+  operations cannot proceed.
+- Administrative connections to the database are blocked, preventing
+  a user from logging in.
+- Monitoring connections to the database are blocked, degrading visibility
+  into the problem.
+- CPU or memory oversubscription due to connection management can lead
+  to a ``mongod`` crashing and restarting. This extra workload on the server
+  can be exclusively attributed to connection handling.
+- In Atlas, connection requests can exceed instance tier limits, causing
+  refusal of new connections.
+
+Avoid Connection Storms
+-----------------------
+
+While no universal solution exists as each problem is unique to a deployment,
+you can use a combination of settings and methods to help you avoid connection
+storms. These include:
+
+- **maxPoolSize**: The maximum number of connections in the connection pool. Beginning
+  with a lower value is recommended to reduce the maximum pressure each server
+  may experience from each connected client applications.
+  |
+  Fewer overall connections in the pool means the database server is using
+  less resources maintaining connections.
+
+- **maxConnecting**: The maximum number of connections the driver can concurrently
+  establish to an individual server before making them available in the connection pool.
```

Comment:
I suggest "The maximum number of new connections the driver will attempt to make to a given server at the same time"

An example may help:

"For example, a maxConnecting setting of 5 will ensure that client threads with new database operations will block if 5 connections are currently being opened for 5 other requests."